src/bindings/python/src/openvino/runtime/opset6/ops.py

```diff
@@ -140,29 +131,28 @@ def read_value(init_value: NodeInput,
     :param name:         Optional name for output node.
     :return: ReadValue node
     """
-    attr_map: Dict[str, Any] = {"variable_id": variable_id}
+    info = VariableInfo()
+    info.variable_id = variable_id
 
     if variable_type is not None:
         if not isinstance(variable_type, Type) and not isinstance(variable_type, str):
-            attr_map["variable_type"] = get_element_type_str(variable_type)
+            info.data_type = get_element_type(variable_type)
         else:
-            attr_map["variable_type"] = variable_type
+            info.data_type = variable_type
 
     if variable_shape is not None:
-        attr_map["variable_shape"] = PartialShape(variable_shape)
+        info.data_shape = PartialShape(variable_shape)
 
-    return _get_node_factory_opset6().create(
-        "ReadValue",
-        [as_node(init_value, name=name)],
-        attr_map,
-    )
+    var_from_info = Variable(info)
+    return _read_value(new_value=as_node(init_value, name=name), variable=var_from_info)
 
 
-@read_value.register
-def _(variable_id: str,
-      variable_type: Optional[Union[NumericType, Type, str]] = None,
-      variable_shape: Optional[TensorShape] = None,
-      name: Optional[str] = None) -> Node:
+@overloading(str, Optional[Union[type, np.dtype, Type, str]], Optional[Union[TensorShape, Shape, PartialShape]], Optional[str])  # type: ignore
+@nameable_op
+def read_value(variable_id: str,  # noqa: F811
+               variable_type: Optional[Union[type, np.dtype, Type, str]] = None,
+               variable_shape: Optional[Union[TensorShape, Shape, PartialShape]] = None,
+               name: Optional[str] = None) -> Node:
```

Comment:
Could you please explain why custom overloading mechanism is needed here and @read_value.register is not sufficient?

Reply:
Because `singledispatch` distinguishes functions based on [the first argument.](https://docs.python.org/3/library/functools.html#functools.singledispatch)
And as I needed to add also new overloadings, they conflicted between each other

Reply:
I see, thanks!